### ISSUE TITLE
Ids for biological SexNames for source and target

### ISSUE THREAD

**Issue by mdrishti** [globalbioticinteractions/globalbioticinteractions#1032]
Hi again @jhpoelen. Continuing on #1031 I also started working on the fourth characteristic - biologoical sex names and ids for source and target organisms. 

So, this is more complicated than the other 3 characteristics. Here https://tinyurl.com/BioSNames , I have a list of values for the name and id for a set of interaction records (record as in one row in interactions.tsv describing one interaction). Now, I understand that the organism(s) in question for a record could occur in clusters or groups and that is why I see values in column ```source/target-SexName``` like "11 males, 4 females, 2 juveniles" or "1 male, 1 female, 5 juveniles" or "1 male 4 female". But, I was wondering if this was the original purpose for this column. If I understand correctly, these values are describing both names and quantities ascribing to each sex. Is that right? If yes, then maybe there should be another column describing the quantity for the number of organisms in  each record, and then such collective values might make sense, only if the column ```source/target-SexId``` also contains appropriate delimited ids. This may also call for further annotation of these columns automatically by GloBI pipeline.

Let me know your thoughts.

**Comment by jhpoelen** [comment]
> Unrelated : While going through the list in the, it is funny how 'female' has been mis-spelled the most out of all the terms :-) . And a lot of spanish terms!

Yes! Lots of variations . . . perhaps an idea to compile common typos and variations to help link the words to their intended terms ? 

| provided term | aligned term |
| --- | --- | 
| femle | female |
| ... | ... |

**Comment by mdrishti** [comment]
@jhpoelen Thanks for your insights. I implemented a reasonable solution incorporating most of what you mentioned in a couple of python functions here: https://github.com/digital-botanical-gardens-initiative/rdflib_trydb_globi/blob/main/src/functions/matchNamesBiologicalGender.py

In the final output, it looks something like the first table in your previous response: 


> source |	interactionType |	target |	targetSexName | targetSexId |
>-----------------------------------------------------------------------------------------------------
> cat |	eats	| mouse	| 11 males, 4 female, 2 juveniles | 11 PATO:0000384, 4 PATO:0000383, 2 PATO:0001190

See also a list here:  https://github.com/digital-botanical-gardens-initiative/rdflib_trydb_globi/blob/main/ontology/data/globi/correctedBiologicalSexNames.tsv, where I mapped all obscure/non-obscure genders listed in GloBI interactions.tsv 

I would love to incorporate this in GloBI. However, I am java-illiterate ;-)! Converting the python code to a standalone tool is one option and then the tool can be called within the GloBI's java code to map. Or maybe just the mapping-list (https://github.com/digital-botanical-gardens-initiative/rdflib_trydb_globi/blob/main/ontology/data/globi/correctedBiologicalSexNames.tsv) can be included in GloBI to map biological genders. 

Let me know your thoughts.
